model-generator.py

```diff
@@ -139,14 +139,47 @@ def apply_overrides(base_data, overrides={}):
                 addr = new_item.get("id")
                 if addr in existing_addresses:
                     print(
-                        f"Warning: id {id} already exists in '{component_type}', skipping."
+                        f"Warning: id {addr} already exists in '{component_type}', skipping."
                     )
                 else:
                     base[component_type].append(new_item)
 
     return base
 
 
+def resolve_inheritance_chain(model_file, override_dir):
+    """
+    Resolve the inheritance chain for a model file.
+    Returns a list of override dictionaries in order from most general to most specific.
+    """
+    # Load current model
+    overrides = load_yaml(model_file)
+
+    # Check if this model inherits from another
+    if "inherits" in overrides:
```

Comment:
The wording `inherits` can be a bit ambiguous, my advice is to change that key name to `parent`, which is clear and also reads as 'one parent and not more then one parent'